test/e2e/kappcontroller/package_test.go

```diff
@@ -409,3 +409,27 @@ spec:
 		}
 	})
 }
+
+func TestPackageNotExists(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	k := e2e.Kubectl{t, env.Namespace, logger}
+	packageName := "foo.1.0.0"
+	expectedError := "stderr: 'Error from server (NotFound): package.data.packaging.carvel.dev \"foo.1.0.0\" not found"
+
+	logger.Section("Get Package", func() {
+		_, err := k.RunWithOpts([]string{"get", "package", packageName}, e2e.RunOpts{AllowError: true})
+		if !strings.Contains(err.Error(), expectedError) {
```

Comment:
Let's check if err is non nil first.
We are trying to switch to `assert`, so let's use that.